### PR TITLE
Update latest after s3 upload and replication

### PR DESCRIPTION
**Pull request by grischard**
Added a bit of error handling on the way

### CODE REVIEW

**Comment by Firefishy** [comment]
Would you mind removing the parallel upload? It doesn't really help if only 1 of the files is large.

File: cookbooks/planet/templates/default/planetdump.erb [hunk 1]
```diff
@@ -2,8 +2,8 @@
 
 # DO NOT EDIT - This file is being maintained by Chef
 
-# Exit on error
-set -e
+# Exit on errors
+set -euo pipefail
```
Comment (Zaczero) [inline cookbooks/planet/templates/default/planetdump.erb:6]: -o pipefail is generally not recommended to be enabled globally. It often results in unexpected and weird pipeline behavior. It's okay for pipelines to return non-zero status codes.

File: cookbooks/planet/templates/default/planetdump.erb [hunk 2]
```diff
@@ -137,6 +137,27 @@ function mk_torrent {
 	"${rss_file}"
 }
 
+function replication_status_check {
+  local s3_url="$1"
+  local retries=100
+  local sleep_duration=10  # 10 seconds, adjust as needed
```
Comment (Zaczero) [inline cookbooks/planet/templates/default/planetdump.erb:143]: 16 minutes deadline seems short

**Comment by grischard** [comment]
I've removed the parallel upload.

File: cookbooks/planet/templates/default/planetdump.erb [hunk 3]
```diff
@@ -146,20 +167,44 @@ function install_dump {
...
-  ln -sf "${year:-.}/${name}" "${dir}/${latest}"
-  test -f "${name}.torrent" && mv "${name}.torrent" "${dir}/${year}" && ln -sf "${year:-.}/${name}.torrent" "${dir}/${latest}.torrent"
-  test -f "${rss_file}" && xmllint --noout "${rss_file}" && cp -f "${rss_file}" "${dir}"
+  # Move the dump and its MD5 checksum to the target directory
+  mv "${name}" "${name}.md5" "${dir}/${year}" || { echo "Error moving files to ${dir}/${year}"; exit 1; }
+  # Create or update the symbolic link for the latest dump
+  ln -sf "${name}" "${dir}/${latest}" || { echo "Error creating symbolic link for ${name}"; exit 1; }
+  # Move torrent file and update its symbolic link
+  mv "${name}.torrent" "${dir}/${year}" || { echo "Error moving ${name}.torrent"; exit 1; }
+  ln -sf "${year:-.}/${name}.torrent" "${dir}/${latest}.torrent" || { echo "Error creating symbolic link for ${name}.torrent"; exit 1; }
```
[the remark this answers is not in the document]
Comment (grischard) [reply cookbooks/planet/templates/default/planetdump.erb:199]: It's very reliable, which makes errors rare, which is why I wanted to have clear error messages here. Situations where things like this fail, for example a full disk, are uncommon and stressful, and I find that good error messages help in unfamiliar situations. I'm happy to remove the error messages if that's the consensus.

[the remark this answers is not in the document]
Comment (Zaczero) [reply cookbooks/planet/templates/default/planetdump.erb:199]: Thank you for sharing that! I just personally like simple solutions :slightly_smiling_face:.

File: cookbooks/planet/templates/default/planetdump.erb [hunk 4]
```diff
@@ -146,20 +167,41 @@ function install_dump {
...
+  # Ensure the target directory exists
   mkdir -p "${dir}/${year}"
-  mv "${name}" "${name}.md5" "${dir}/${year}"
-  ln -sf "${year:-.}/${name}" "${dir}/${latest}"
-  test -f "${name}.torrent" && mv "${name}.torrent" "${dir}/${year}" && ln -sf "${year:-.}/${name}.torrent" "${dir}/${latest}.torrent"
-  test -f "${rss_file}" && xmllint --noout "${rss_file}" && cp -f "${rss_file}" "${dir}"
+  # Move the dump and its MD5 checksum to the target directory
+  mv "${name}" "${name}.md5" "${dir}/${year}" || { echo "Error moving files to ${dir}/${year}"; exit 1; }
+  # Create or update the symbolic link for the latest dump
+  ln -sf "${name}" "${dir}/${latest}" || { echo "Error creating symbolic link for ${name}"; exit 1; }
```
Comment (Zaczero) [inline cookbooks/planet/templates/default/planetdump.erb:196]: why ${year:-.}/ gone?

File: cookbooks/planet/templates/default/planetdump.erb [hunk 5]
```diff
@@ -137,6 +137,27 @@ function mk_torrent {
 	"${rss_file}"
 }
 
+function replication_status_check {
+  local s3_url="$1"
+  local retries=100
+  local sleep_duration=10  # 10 seconds, adjust as needed
+
+  for ((i=1; i<=retries; i++)); do
+    local replication_status=$(curl -sI "${s3_url}" | grep -F 'x-amz-replication-status' | awk '{print $2}' | tr -d '\r\n')
```
Comment (Zaczero) [inline cookbooks/planet/templates/default/planetdump.erb:146]: https://www.shellcheck.net/wiki/SC2155
```suggestion
    local replication_status
    replication_status=$(curl -sI "${s3_url}" | grep -F 'x-amz-replication-status' | awk '{print $2}' | tr -d '\r\n')
```

**Comment by Zaczero** [comment]
I have implemented some suggestions for easy copy-pasting:
https://gist.github.com/Zaczero/3101ea54051cfab2aea5f5a00be71647

See the diff for additional details.

File: cookbooks/planet/templates/default/planetdump.erb (hunk 4 above)
Comment (grischard) [reply cookbooks/planet/templates/default/planetdump.erb:196]: As far as I can tell, `${year:-.}` is always empty here?

Comment (tomhughes) [reply cookbooks/planet/templates/default/planetdump.erb:196]: If it then something has changed - the two PBF calls to that function don't set it currently as they don't files by year in the local store.

**Comment by Firefishy** [comment]
Merged a flavour of this via: https://github.com/openstreetmap/chef/pull/636

**Comment by Firefishy** [comment]
Thank you for the PR

**Pull request closed without merging**